### PR TITLE
changed port/HTTP version parts to Ints

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

- `Int` is the currency type for integral things and we already sort of
  changed the the type for ports but not everywhere.
- generic parameters that aren't just 'any type' shouldn't be just `T`
  or `S` or something

Modifications:

- make more use of `Int`
- rename generic parameters to something more descriptive like `Bytes`

Result:

easier to read/write NIO code

### CODE REVIEW

File: Sources/NIO/SocketAddresses.swift [hunk 1]
```diff
@@ -126,12 +126,12 @@ public enum SocketAddress: CustomStringConvertible {
     }
 
     /// Get the port associated with the address, if defined.
-    public var port: UInt16? {
+    public var port: Int? {
         switch self {
         case .v4(let addr):
-            return UInt16(bigEndian: addr.address.sin_port)
+            return Int(UInt16(bigEndian: addr.address.sin_port))
```
Comment (Lukasa) [inline Sources/NIO/SocketAddresses.swift]: Why is this round-tripping through `UInt16`?

Comment (weissi) [reply Sources/NIO/SocketAddresses.swift]: @Lukasa step 1) change the endianness (only available within the same type), step 2) change the type

Comment (Lukasa) [reply Sources/NIO/SocketAddresses.swift]: Can you add a comment to that effect please? People will definitely forget.

Comment (weissi) [reply Sources/NIO/SocketAddresses.swift]: yes, done

**Pull request merged**